go-controller/pkg/ovn/pods.go

```diff
@@ -613,6 +610,27 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) (err error) {
 	return nil
 }
 
+func (oc *Controller) updatePodAnnotationWithRetry(origPod *kapi.Pod, podInfo *util.PodAnnotation) error {
+	resultErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
```

Comment:
not really, today, we just directly set the annotation, but only for the "k8s.ovn.org/pod-networks" annotation. since it is not for multi-homing, it is expected that there is no other party to update this annotation at the same time, so there is no race anywhere.

now, since the same key "k8s.ovn.org/pod-networks" is shared by all the network controller, they could try to update the same annotation at the same time, so we have to use retryOnConflict() and updatePod(). In the case of cocurrent update, there could be conflict now, and we'd have to get the newly updated pod and update again until conflict settles.

Reply:
I see. But what if some other component (not ovn-k) tries to update any annotation on the pod at the same time ? We'd run into this situation, right ? 

Never the less, your answer above clearly explains the reasons. Thank you.

Reply:
OK, so, even if we somehow manage to find a way to coalesce all the network controllers into a single one, this would still potentially have conflicts.

I guess this has to be there, independently of follow up commits.